packages/shared/src/components/PostOptionsReadingHistoryMenu.tsx

```diff
@@ -0,0 +1,156 @@
+      const modified = readHistory.edges.map(({ node }) => {
+        if (node.post.id === id) {
+          node.post = {
+            ...node.post,
+            ...update(node.post),
+          };
+        }
+        return { node };
+      });
+      readHistory.edges = modified;
+      return {
+        readHistory,
+      };
+    });
+    queryClient.setQueryData<ReadHistoryInfiniteData>(historyQueryKey, {
+      ...oldReadingHistory,
+      pages: newItems,
+    });
+    return () => {
+      queryClient.setQueryData<ReadHistoryInfiniteData>(
+        historyQueryKey,
+        oldReadingHistory,
+      );
+    };
+  };
+/* eslint-disable no-param-reassign */
+
+export default function PostOptionsReadingHistoryMenu({
+  post,
+  onHidden,
+  onHide,
+}: PostOptionsReadingHistoryMenuProps): ReactElement {
+  const { user } = useContext(AuthContext);
+  const queryClient = useQueryClient();
+  const historyQueryKey = ['readHistory', user?.id];
+
+  /* eslint-disable @typescript-eslint/no-unused-vars */
+  const [isPostLinkCopied, copyPostLink] = useCopyLink(
+    () => post.commentsPermalink,
+  );
+
+  const { bookmark, removeBookmark } = useBookmarkPost({
+    onBookmarkMutate: updateReadingHistoryPost(
+      queryClient,
+      historyQueryKey,
+      () => ({
+        bookmarked: true,
+      }),
+    ),
+    onRemoveBookmarkMutate: updateReadingHistoryPost(
+      queryClient,
+      historyQueryKey,
+      () => ({
+        bookmarked: false,
+      }),
+    ),
+  });
+
+  const onBookmarkReadingHistoryPost = async (): Promise<void> => {
+    if (post?.bookmarked) {
+      return removeBookmark(post);
+    }
+    return bookmark(post);
+  };
+
+  const getBookmarkIcon = () => {
+    if (post?.bookmarked) {
+      return <MenuIcon Icon={BookmarkIconFilled} />;
+    }
+    return <MenuIcon Icon={BookmarkIcon} />;
+  };
+
+  const getBookmarkMenuText = () => {
+    if (post?.bookmarked) {
+      return 'Remove from bookmarks';
+    }
+    return 'Save to bookmarks';
+  };
+
+  return (
+    <>
+      <PortalMenu
+        disableBoundariesCheck
+        id="reading-history-options-context"
+        className="menu-primary"
+        animation="fade"
+        onHidden={onHidden}
+      >
+        <Item key="share" className="typo-callout" onClick={copyPostLink}>
+          <a className="flex w-full typo-callout">
+            <MenuIcon Icon={ShareIcon} /> Share article
+          </a>
+        </Item>
+        <Item
+          key="bookmarks"
+          className="typo-callout"
+          onClick={onBookmarkReadingHistoryPost}
+        >
+          <a className="flex w-full typo-callout">
+            {getBookmarkIcon()}
+            {getBookmarkMenuText()}
+          </a>
+        </Item>
+        <Item
+          key="close"
+          className="laptop:hidden typo-callout"
+          onClick={() => onHide(post?.id)}
+        >
+          <a className="flex w-full typo-callout">
+            <MenuIcon Icon={XIcon} /> Remove article
```

Comment:
When using the new icons the idea is actually that they change on certain states, if we don't use that yet I would actually suggest sticking to the old ones.

Reply:
Should I use the old x icon even if there is the new one in the design guide?